services/catalog/openapi.json

```diff
@@ -961,39 +939,70 @@
               "type": "string",
               "format": "uuid"
             },
-            "description": "Used in group-nodes. Node IDs of those connected to the output",
-            "example": [
-              "nodeUuid1",
-              "nodeUuid2"
-            ]
+            "description": "Used in group-nodes. Node IDs of those connected to the output"
           },
           "parent": {
-            "title": "Parent",
-            "type": "string",
-            "description": "Parent's (group-nodes') node ID s. Used to group",
-            "format": "uuid",
-            "example": [
-              "nodeUUid1",
-              "nodeUuid2"
+            "anyOf": [
+              {
+                "type": "null"
+              },
+              {
+                "title": "Parent",
+                "description": "Parent's (group-nodes') node ID s. Used to group",
+                "type": "string",
+                "format": "uuid"
+              }
             ]
           },
+          "position": {
+            "title": "Position",
+            "allOf": [
+              {
+                "$ref": "#/components/schemas/Position"
+              }
+            ],
+            "deprecated": true
+          },
           "state": {
+            "title": "State",
             "allOf": [
               {
-                "$ref": "#/components/schemas/RunningState"
+                "$ref": "#/components/schemas/NodeState"
               }
             ],
-            "description": "the node's running state",
-            "default": "NOT_STARTED"
+            "description": "The node's state object"
+          }
+        },
+        "additionalProperties": false
+      },
+      "NodeState": {
+        "title": "NodeState",
+        "type": "object",
+        "properties": {
+          "modified": {
+            "title": "Modified",
+            "type": "boolean",
+            "description": "true if the node's outputs need to be re-computed",
+            "default": true
           },
-          "position": {
-            "title": "Position",
+          "dependencies": {
+            "title": "Dependencies",
+            "uniqueItems": true,
+            "type": "array",
+            "items": {
+              "type": "string",
+              "format": "uuid"
+            },
+            "description": "contains the node inputs dependencies if they need to be computed first"
```

Comment:
Can we have an empty array as default?

Reply:
YEs, this is possible. But I would ask @sanderegg to defined on that field, i.e. ``NodeState.dependencies`` logic.